packages/common-all/src/engine/EngineV3Base.ts

```diff
@@ -382,4 +392,88 @@ export abstract class EngineV3Base implements ReducedDEngine {
       })
     );
   }
+
+  /**
+   * Create backlink from given link that references another note (denoted by presence of link.to field)
+   * and add that backlink to referenced note's links
+   *
+   * @param link Link potentionally referencing another note
+   */
+  protected async addBacklink(link: DLink): Promise<NoteChangeEntry[]> {
+    const maybeBacklink = BacklinkUtils.createFromDLink(link);
+    if (maybeBacklink) {
+      const maybeVault = link.to?.vaultName
+        ? VaultUtils.getVaultByName({
+            vname: link.to?.vaultName,
+            vaults: this.vaults,
+          })
+        : undefined;
+      const notes = await this.noteStore.findMetaData({
+        fname: link.to!.fname!,
+        vault: maybeVault,
+      });
+      if (notes.data) {
+        return Promise.all(
+          notes.data.map(async (note) => {
+            const prevNote = _.merge(_.cloneDeep(note), {
+              body: "",
+            });
+            BacklinkUtils.addBacklink({ note, backlink: maybeBacklink });
+            // Temp solution to get around current restrictions where NoteChangeEntry needs a NoteProp
+            const updatedNote = _.merge(note, {
+              body: "",
+            });
+            return {
+              prevNote,
+              note: updatedNote,
+              status: "update",
+            };
+          })
+        );
+      }
+    }
+    return [];
+  }
+
+  /**
+   * Remove backlink associated with given link that references another note (denoted by presence of link.to field)
+   * from that referenced note
+   *
+   * @param link Link potentionally referencing another note
+   */
+  protected async removeBacklink(link: DLink): Promise<NoteChangeEntry[]> {
```

Comment:
aside - should we maybe undeprecate `DLink`? It's been marked as deprecate for as long as I can remember and its suggested replacement `DNoteLink` is practically the same and that interface needs some cleanup too.  @kevinslin

Reply:
no, we should continue to remove it

we use `DNoteLink` in new methods, the key difference is the location of the `alias` field (see references of dlink2DNoteLink)
    - DLink has an `alias` field, it's not supposed to be on the link field but actually in the `from/to` field
    - `DNoteLoc` is a super set of `DLoc` (has an `alias` field)
    - `DNoteLink` is used in other places